### PR TITLE
Remove stage 2

### LINKED ISSUE #2: Is Phase 2 (ProcessingMediaStreamTrack) really needed?
In implementing the Breakout Box, the team has found it easier to implement TrackProcessor and TrackGenerator as separate object than to implement the compound object "ProcessingMediaStreamTrack".

This calls into question whehter it is needed at all; given that it can be easily shimmed, should we remove it from the specification?

### PR DESCRIPTION
**Pull request by alvestrand**
Fixes #2 per WG discussion

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/mediacapture-insertable-streams/pull/7.html" title="Last updated on Jan 19, 2021, 8:08 AM UTC (59c11cd)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/mediacapture-insertable-streams/7/d6bf42b...59c11cd.html" title="Last updated on Jan 19, 2021, 8:08 AM UTC (59c11cd)">Diff</a>

### CODE REVIEW

File: explainer.md [hunk 1]
```diff
@@ -35,22 +35,39 @@ The use cases include:
...
+<pre>
+originalTrack = navigator.getUserMedia(...)
+myProcessingElement = new TransformStream(....)
+myTrackProcessor = new TrackProcessor(originalTrack);
+myTrackGenerator = new TrackGenerator();
+
+myTrackProcessor.readable.pipeThrough(myProcessingElement).pipeTo(myTrackGenerator.writable);
+myTrackGenerator.readable.pipeTo(myTrackProcessor.writable);
+
+pc.addtrack(myTrackGenerator);
```
Comment (alvestrand) [inline explainer.md:62]: capital T

File: explainer.md [hunk 2]
```diff
@@ -35,22 +35,39 @@ The use cases include:
 * Background removal
 * Voice processing
 
+## Breakout Box Properties
+
+The Breakout Box conceptually splits the MediaStreamTrack into two components:
+* The TrackProcessor, which consumes a MediaStreamTrack's source and generates a stream of
+media frames (VideoFrame or AudioFrame)
+* The TrackGenerator, which consumes a stream of media frames and exposes a MediaStreamTrack
+interface, so that it can be used anywhere a MediaStreamTrack is currently attched.
```
Comment (henbos) [inline explainer.md:44]: attached

File: explainer.md [hunk 3]
```diff
@@ -35,22 +35,39 @@ The use cases include:
...
+
+There is no connection between the stream generator and stream consumer that doesn't pass
+through Javascript; one can freely interconnect the streams in whatever fashion one
```
Comment (henbos) [inline explainer.md:47]: JavaScript?

File: explainer.md [hunk 4]
```diff
@@ -35,22 +35,39 @@ The use cases include:
...
+sees fit.
 
 ## Code Examples
 
+Example of processing media before transmitting to a PeerConnection in Phase Two:
+<pre>
+originalTrack = navigator.getUserMedia(...)
+myProcessingElement = new TransformStream(....)
```
Comment (henbos) [inline explainer.md:55]: ...?

File: explainer.md (hunk 1 above)
Comment (henbos) [inline explainer.md:62]: addTrack

File: explainer.md [hunk 5]
```diff
@@ -81,6 +98,17 @@ enum ControlSignalName {
 };
 </pre>
 
+Note that despite using the same names for the streams defined, the meaning is the opposite
+to the same attributes in a TransformStream; these are interfaces you plug a TransformStream
+into, not interfaces you use to transform something.
+
+<b>Alternative approach</b>: use function calls for createWritable and createReadable that return
+the created streams
+.
```
Comment (henbos) [inline explainer.md:107]: streams.

**Review by henbos — CHANGES_REQUESTED** [review]

File: explainer.md (hunk 4 above)
Comment (alvestrand) [reply explainer.md:55]: {init() {... }, {transform(frame, controller) { ... }} seemed like a bit excessive to write down, given that this is defined by the TransformStream spec.

**Pull request merged**